cns/kubecontroller/nodenetworkconfig/conversion.go

```diff
@@ -87,13 +91,38 @@ func CreateNCRequestFromStaticNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwor
 
 	secondaryIPConfigs := map[string]cns.SecondaryIPConfig{}
 
-	// iterate through all IP addresses in the subnet described by primaryPrefix and
-	// add them to the request as secondary IPConfigs.
-	for addr := primaryPrefix.Masked().Addr(); primaryPrefix.Contains(addr); addr = addr.Next() {
-		secondaryIPConfigs[addr.String()] = cns.SecondaryIPConfig{
-			IPAddress: addr.String(),
-			NCVersion: int(nc.Version),
+	switch nc.Type {
+	case v1alpha.VNETBlock:
+		{
+			for blockIndex, ipAssignment := range nc.IPAssignments {
+				cidrPrefix, err := netip.ParsePrefix(ipAssignment.IP)
+				if err != nil {
+					return nil, errors.Wrapf(ErrInvalidCidrIPBlock, "CIDR IP Block: %s", ipAssignment.IP)
+				}
+				// iterate through all IP addresses in the CIDR block described by cidrPrefix and
+				// add them to the request as secondary IPConfigs.
+				for addr := cidrPrefix.Masked().Addr(); cidrPrefix.Contains(addr); addr = addr.Next() {
+					if blockIndex == 0 {
+						addr = addr.Next() // # Block 1 CIDR. Exclude 1st IP from pod assignment
```

Comment:
I think this check belongs outside the inner for-loop, otherwise you are skipping more than the first IP on the first block (besides checking the condition redundantly). for that matter, initializing "addr" explicitly before the loop seems more readable if we need to handle at least this special case.

Reply:
fixed